lib/APPTurtle/src/CustomRosTransport.cpp

```diff
@@ -63,81 +75,111 @@
 
 bool CustomRosTransport::open(uxrCustomTransport* transport)
 {
-    bool isOpen = false;
+    CustomRosTransport * tp = toThis(transport);
+    return (nullptr != tp) ? tp->open() : false;
+}
+
+bool CustomRosTransport::close(uxrCustomTransport* transport)
+{
+    CustomRosTransport * tp = toThis(transport);
+    return (nullptr != tp) ? tp->close() : false;
+}
+
+size_t CustomRosTransport::write(uxrCustomTransport* transport, const uint8_t* buffer, size_t size, uint8_t* errorCode)
+{
+    CustomRosTransport * tp = toThis(transport);
+    return (nullptr != tp) ? tp->write(buffer, size, errorCode) : 0U;
+}
 
+size_t CustomRosTransport::read(uxrCustomTransport* transport, uint8_t* buffer, size_t size, int timeout,
+                                uint8_t* errorCode)
+{
+    CustomRosTransport * tp = toThis(transport);
+    return (nullptr != tp) ? tp->read(buffer, size, timeout, errorCode) : 0U;
+}
+
+/******************************************************************************
+ * Local Functions
```

Comment:
Local functions are ususally behind the method sections.

Reply:
done